lib/Conversion/TTNNToEmitC/PopulatePatterns.h

```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TTMLIR_LIB_CONVERSION_TTNNTOEMITC_POPULATEPATTERNS_H
+#define TTMLIR_LIB_CONVERSION_TTNNTOEMITC_POPULATEPATTERNS_H
+
+#include "mlir/Dialect/EmitC/IR/EmitC.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+using namespace mlir;
+
+namespace {
+
+template <typename SrcOp, typename Adaptor = typename SrcOp::Adaptor>
+class DefaultOpConversionPattern : public OpConversionPattern<SrcOp> {
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
+
+public:
+  // Default op conversion pattern, used to convert most ops
+  //
+  DefaultOpConversionPattern(MLIRContext *ctx)
+      : OpConversionPattern<SrcOp>(ctx) {}
+
+  DefaultOpConversionPattern(const TypeConverter &typeConverter,
+                             MLIRContext *context, PatternBenefit benefit = 1)
+      : OpConversionPattern<SrcOp>(typeConverter, context, benefit) {}
+
+  // Coverts op name by removing the dialect prefix ("ttnn.") and replacing with
+  // namespace prefix ("ttnn::")
+  //
+  std::string convertOpName(SrcOp op) const {
+    auto name = op.getOperationName();
+    if (name.starts_with("ttnn.")) {
+      return "ttnn::" + name.drop_front(5).str();
+    }
+    return "ttnn::" + name.str();
+  }
+
+  LogicalResult
+  matchAndRewrite(SrcOp srcOp, Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    SmallVector<Attribute, 4> templateArguments;
```

Comment:
Why 4?

Reply:
4 would be the number of elements that `SmallVector` will stack allocate before falling back to the heap.  4 seems like a good number because I doubt many functions have more than 4 templates args.

Reply:
Yeah I know how it works, I was just expecting to see defined const if possible, so that it is obvious why some value is chosen.